features/step_definitions/receive-steps.js

```diff
@@ -16,3 +16,22 @@ Then('I should see {int} addresses', async function (numberOfAddresses) {
   const addressesFound = await getVisibleElementsCountForSelector(this.client, '.WalletReceive_walletAddress', `.generatedAddress-${numberOfAddresses}`);
   expect(addressesFound).to.equal(numberOfAddresses);
 });
+
+Then('I should see the following addresses:', async function (table) {
+  const expectedAdresses = table.hashes();
+  let addresses;
+
+  await this.client.waitUntil(async () => {
+    addresses = await this.client.getAttribute('.WalletReceive_walletAddress','class');
+    return addresses.length === 3;
```

Comment:
This hardcoded number is not good 😉

Reply:
And i would write the whole expectation differently … more like this:

Reply:
@DominikGuzei thanks. very well spotted!